src/VkBootstrap.h

```diff
@@ -195,6 +195,124 @@ struct SystemInfo {
 class InstanceBuilder;
 class PhysicalDeviceSelector;
 
+struct ExtensionFeatures {
+
+    using DeleteProc = void(*)(ExtensionFeatures&);
```

Comment:
The main reason is to try and keep the level of meta required for reflection as thin and as close to the type itself as possible.

If all three were to be static functions, they would need to be templated to ensure copying/updating/deleting is proper. This introduces a can of worms as the type/user/both would need a method of obtaining the type info in order to know which specialization to invoke. This is complicated even further for destruction.

Because all three are function pointers generated from lambdas built around T in the build function, all type info is baked into the procs beforehand, so no type info is needed when provoking copy/updating/destruction semantics. In the eyes of the type, handling these things is nothing more than calling a function pointer that does everything for it.

Reply:
Closing as its no longer relevant since the removal of the function pointers and introduction of FeaturesContainer.